### ISSUE TITLE
Allow limiting number of follows in a role policy

### ISSUE THREAD

**Issue by zimmerwald** [syuilo/Misskey#15146]
### Summary

I propose a new policy a server admin could add to a role which limits the number of people a user can follow to a certain amount.

Inspired by mixi2's (yes you're probably tired of hearing it at this point, sorry) 2000 follow limit: https://bsky.app/profile/brandon.insertcredit.com/post/3ldfxfcdess2f

We can be a lot flexible than mixi2 in this regard though, thanks to the role feature.

### Purpose

New, spammy accounts mass-following users in order to get a lot of follow backs in return has been a problem in SNS; I see it being a regular complaint in Bluesky for example. I am not sure about misskey.io, but it would be very useful for the rest of the Misskey instances that have open registrations. One for example could create a conditional role containing a policy that limits the user to following a maximum of 100 people, which would be removed a week later.

### Do you want to implement this feature yourself?

- [ ] Yes, I will implement this by myself and send a pull request

**Comment by eternal-flame-AD** [comment]
I would love this feature too and am willing to send in a PR if needed. 

There is one thing is the current follower count seem to be off in certain remote-unfollow scenarios at least for me for some reason, so if we want a precise limit it may need a little debugging or refactoring.

私もこの機能がほしいですし、必要あれば PR を送信したいと思います。

1 つだけ小さな問題が、、少なくとも私の場合、何らかの現在のフォロワー数が特定のリモート Unfollow でずれているようです。正確な制限が必要な場合は、少しデバッグまたはリファクタリングが必要になる可能性があります。

**Comment by temtemy** [comment]
I don't think we need to limit the number of users following your account (though I suppose it could be a separate issue since I can see it being abused in order to DoS a Misskey instance by forcing it to subscribe to too many remote instances). We just need to limit the number of users an account can follow (which will always be accurate in the local side where it matters) for this issue.
